kadena-signing-api/src/Kadena/SigningApi.hs

```diff
@@ -45,12 +45,15 @@ this will be the account used to pay the transaction's gas price.
 -- contract (as it stands) appears to be that accounts can't be an empty string
 mkAccountName :: Text -> Either Text AccountName
 mkAccountName n =
-  if not isValidCharset then Left "Invalid Character detected. Must be Latin1, no spaces, control characters or '|'"
-  else if not isCorrectSize then Left "Incorrect length. Must be between 3 and 256 characters in length."
+  if not (isValidCharset n) then Left "Invalid Character detected. Must be Latin1, no spaces, control characters or '|'"
+  else if not (isCorrectSize n) then Left "Incorrect length. Must be between 3 and 256 characters in length."
   else Right $ AccountName n
-  where
-    isValidCharset = T.foldl' (\b c -> b && isValidAccountNameCharacter c) True n
-    isCorrectSize = let l = T.length n in l >= 3 && l <= 256
+
+isCorrectSize :: Text -> Bool
+isCorrectSize n = let l = T.length n in l >= 3 && l <= 256
+
+isValidCharset :: Text -> Bool
+isValidCharset = T.foldl' (\b c -> b && isValidAccountNameCharacter c) True
```

Comment:
Just realized I somehow never used folds on `Text`. Why not `T.all` ?

Reply:
Because I forgot about it.. *ahem*